internal/statemachine/classic_states.go

```diff
@@ -504,84 +513,68 @@ func (stateMachine *StateMachine) manualCustomization() error {
-			imageOpts.Snaps, imageOpts.SnapChannels, err = parseSnapsAndChannels(
-				classicStateMachine.commonFlags.Snaps)
-			if err != nil {
-				return err
-			}
-
-			// If the rootfs has already been pre-seeded, we need to delete the
-			// pre-seeded snaps and redo the preseed with all of the snaps
-			stateFile := filepath.Join(classicStateMachine.tempDirs.rootfs,
-				"var", "lib", "snapd", "seed", "seed.yaml")
-			if _, err := os.Stat(stateFile); err == nil {
-				// check for an existing model assertion file, otherwise snapd will use
-				// a generic model assertion
-				modelFile := filepath.Join(classicStateMachine.tempDirs.rootfs,
-					"var", "lib", "snapd", "seed", "assertions", "model")
-				if _, err := os.Stat(modelFile); err == nil {
-					// create a copy of the model file because it will be deleted soon
-					newModelFile := filepath.Join(classicStateMachine.stateMachineFlags.WorkDir,
-						"model")
-					if err := osutilCopyFile(modelFile, newModelFile, 0); err != nil {
-						return fmt.Errorf("Error copying modelFile from preseeded filesystem: %s",
-							err.Error())
-					}
-					imageOpts.ModelFile = newModelFile
-				}
+	var err error
+	imageOpts.Snaps, imageOpts.SnapChannels, err = parseSnapsAndChannels(classicStateMachine.Snaps)
+	if err != nil {
+		return err
+	}
 
-				// Now remove all of the seeded snaps
-				preseededSnaps, err := removePreseeding(
-					classicStateMachine.tempDirs.rootfs)
-				if err != nil {
-					return fmt.Errorf("Error removing preseeded snaps from existing rootfs: %s",
-						err.Error())
-				}
-				for snap, channel := range preseededSnaps {
-					// if a channel is specified on the command line for a snap that was already
-					// preseeded, use the channel from the command line instead of the channel
-					// that was originally used for the preseeding
-					if _, found := imageOpts.SnapChannels[snap]; !found {
-						imageOpts.Snaps = append(imageOpts.Snaps, snap)
-						imageOpts.SnapChannels[snap] = channel
-					}
-				}
-			}
+	// add any extra snaps from the image definition to the list
+	for _, extraSnap := range classicStateMachine.ImageDef.Customization.ExtraSnaps {
+		if !helper.SliceHasElement(classicStateMachine.Snaps, extraSnap.SnapName) {
+			imageOpts.Snaps = append(imageOpts.Snaps, extraSnap.SnapName)
+		}
+		if extraSnap.Channel != "" {
+			imageOpts.SnapChannels[extraSnap.SnapName] = extraSnap.Channel
+		}
+	}
 
-			imageOpts.Classic = true
-			imageOpts.Architecture = classicStateMachine.Opts.Arch
-			if imageOpts.Architecture == "" {
-				imageOpts.Architecture = getHostArch()
-			}
+	// plug/slot sanitization not used by snap image.Prepare, make it no-op.
+	snap.SanitizePlugsSlots = func(snapInfo *snap.Info) {}
 
-			imageOpts.PrepareDir = classicStateMachine.tempDirs.rootfs
+	// iterate through the list of snaps and ensure that all of their bases
+	// are also set to be installed
+	for _, seededSnap := range imageOpts.Snaps {
+		snapStore := store.New(nil, nil)
```

Comment:
using a Store directly like this will not work with anything but the main store, ToolingStore exist for that, otoh it doesn't support SnapInfo but could